android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/QuestionnaireExtension.kt

```diff
@@ -181,5 +181,6 @@ fun String.castToType(type: DataType): Type? {
     DataType.URI -> UriType(this)
     DataType.CODING -> this.tryDecodeJson<Coding>()
     DataType.QUANTITY -> this.tryDecodeJson<Quantity>()
+    else -> null /*TODO cast the (several) remaining Enumeration.DataTypes*/
```

Comment:
is null OK or should we throw here?

Reply:
We'd update the implementation here (maybe during CHT) to handle rest of the relevant DataTypes but for now returning `null` is okay.